packages/nimble-components/src/theme-provider/index.ts

```diff
@@ -33,19 +53,50 @@ export const theme = DesignToken.create<Theme>({
  * @internal
  */
 export class ThemeProvider extends FoundationElement {
-    @attr({
-        attribute: 'direction'
-    })
+    @attr()
+    public override lang = '';
+
+    @attr()
     public direction: Direction = Direction.ltr;
 
-    @attr({
-        attribute: 'theme'
-    })
+    @attr()
     public theme: Theme = Theme.light;
 
+    public get validity(): ValidityObject {
+        return {
+            invalidLang: this.langIsInvalid
+        };
+    }
+
+    private langIsInvalid = false;
+
+    public checkValidity(): boolean {
+        return !this.langIsInvalid;
+    }
+
+    public langChanged(
+        _prev: string | undefined | null,
+        next: string | undefined | null
+    ): void {
+        this.langIsInvalid = false;
+        let shouldSetNewTokenValue = false;
+        if (next) {
+            this.langIsInvalid = isInvalidLang(next);
+            if (!this.langIsInvalid) {
+                shouldSetNewTokenValue = true;
+            }
+        }
+
+        if (shouldSetNewTokenValue) {
+            lang.setValueFor(this, next!);
+        } else {
+            lang.deleteValueFor(this);
+        }
```

Comment:
This boolean flag tracking is pretty confusing. Think it can be simplified just making sure each code path updates the validity and token value as needed.

```ts
if (next === null || next === undefined) {
    lang.deleteValueFor(this);
    this.langIsInvalid = false;
    return;
}


if (isInvlaidLang(next)) {
    lang.deleteValueFor(this);
    this.langIsInvalid = true;
} else {
    lang.setValueFor(this, next);
    this.langIsInvalid = false;
}
```

Reply:
Actually based on the other thread I'm not sure if in the isInvalidLang(next) should result in clearing the token (i.e. falling back to the cascade).

The other thread:

>> If set to invalid value, resolve to parent?
>
> No. The value remains that specific invalid value. It is up to consumers of the locale to determine how to handle an invalid value.

Says the DOM value does not fallback to the cascade at that point. Trying to understand if there is a con for falling back to the cascade.

Reply:
Based on the conversation in the other thread I thought we had decided that our token value would not match the DOM behavior, i.e. our token would not take on invalid/unknown values. Are we on the same page? If yes, then the logic will stay the same (except I can refactor a bit for clarity).

Reply:
Okay, talked on Teams -- we're going to fall back to the cascade, then if there are no valid values in the cascade, the token will try to use the documentElement's `lang` value, then if that is invalid/missing, we'll fall back to "en-US".

Reply:
Unresolving, please update to a pattern similar to the first comment in the thread instead of the `houldSetNewTokenValue` pattern

Reply:
Sorry, in all the discussion about our fallback behavior, I lost sight of the original ask.